### ISSUE TITLE
Running trainer.test on val_dataloader hangs program

### ISSUE THREAD

**Issue by surya-narayanan** [PyTorchLightning/pytorch-lightning#8123]
## 🐛 Bug

I run a test loop on my val data loader after training with a checkpoint loaded from disk to report the best performance on the validation set. However, some times, this hangs and when I kill the program, a few jobs are left hanging running python.

### To Reproduce

I am unable to reproduce it using the Boring Model. 
https://colab.research.google.com/drive/1hlnuOxplfIk8rAsg4TbOEDHo1iL58Gj4?usp=sharing

I could not reproduce the bug using the Boring model. This is because it happens some times, not always. I thought it could be because I am reading images from disk during training, but I checked and am not. 

### Expected behavior

Works fine and reports statistics on val data loader

### Environment
* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.2
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.9.0+cu102
	- pytorch-lightning: 1.3.7post0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.10
	- version:           #1 SMP Sat Jun 5 09:50:34 PDT 2021

**Comment by awaelchli** [comment]
> I run a test loop on my val data loader after training with a checkpoint loaded from disk to report the best performance on the validation set.


Hi @surya-narayanan 
still experiencing this issue? Could you share a few details how you saved the checkpoints (model chekcpoint callback?) and how you are loading that model back for testing.

**Comment by awaelchli** [comment]
Closing for now, not reproducible at this moment.
